baxter_collaboration_lib/test/test_gripper.cpp

```diff
@@ -1,27 +1,218 @@
 #include <gtest/gtest.h>
-
 #include "robot_interface/gripper.h"
+#include <iostream>
 
 using namespace std;
 
-TEST(GripperTest, testConstructorDefaultValues)
+// subscribes to the grippers command
+// and publishes to the grippers state
+class gripperTester
 {
-    string    limb = "left";
-    bool use_robot =  false;
+    ros::NodeHandle             nh;
+    std::string               name;
+    std::string               limb;
+    ros::Publisher        prop_pub;
+    ros::Publisher       state_pub;
+    ros::Subscriber        cmd_sub; 
+    std::mutex    mutex_properties; 
+    ros::AsyncSpinner      spinner;
 
-    Gripper gr(limb, use_robot);
+public:
+    explicit gripperTester(std::string _name, std::string _limb) : 
+                                name(_name), limb(_limb), spinner(1)
+    {
+        prop_pub = nh.advertise<baxter_core_msgs::EndEffectorProperties>(
+                                 "/robot/end_effector/" + limb + "_gripper/properties", 1, true);
+
+        state_pub = nh.advertise<baxter_core_msgs::EndEffectorState>(
+                                 "/robot/end_effector/" + limb + "_gripper/state", 1, true);
+   
+        cmd_sub = nh.subscribe("/robot/end_effector/" + limb + "_gripper/command",
+                                 SUBSCRIBER_BUFFER, &gripperTester::commandCb, this);
+        
+        spinner.start();
+
+        // sleep to wait for the publishers to be ready
+        ros::Duration(.5).sleep();
+    }
+
+    void commandCb(const baxter_core_msgs::EndEffectorCommand &msg)
+    {
+       // std::string _msg = &msg.command;
+       // std::string _cmd;
+
+
+        baxter_core_msgs::EndEffectorState state;
+        
+        state.calibrated = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.enabled    = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.error      = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.gripping   = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.missed     = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.ready      = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+        state.moving     = baxter_core_msgs::EndEffectorState::STATE_UNKNOWN;
+       
+
+        baxter_core_msgs::EndEffectorCommand _cmd;
+        _cmd.command = baxter_core_msgs::EndEffectorCommand::CMD_CALIBRATE;
+
+        if (msg.command == _cmd.command)
+        {
+            state.calibrated = baxter_core_msgs::EndEffectorState::STATE_TRUE;
+        } 
+
+        state_pub.publish(state);
+    }
+
+    // Publishes properties of the suction cup gripper
+    void sendPropertiesSuction()
+    {
+        //publish the properties then make sure they are correctly set in the tests
+        //std::lock_guard<std::mutex> lock(mutex_properties);
+        baxter_core_msgs::EndEffectorProperties prop;
+        prop.ui_type = 1u;
```

Comment:
could comparision between unsigned and signed integers be causing an issue with the startup properties?

Also note that the properties inside each Gripper object are set once as per the latest commit. After this happens the subscriber to the properties is shut down.